rules.go

```diff
@@ -225,15 +225,38 @@ func gocriticArgOrder(m fluent.Matcher) {
 		Suggest(`strings.Contains($s2, $s1)`)
 }
 
+func badSplitNValue(m fluent.Matcher) {
+	// Interpret any non-positive match count as "match all" and replace it by a better function.
+	// The special case is zero. Zero does nothing, so it's a bug, so we replace it too.
+	// Everything else is for consistency.
+	m.Match(`strings.Replace($s, $d, $w, $n)`).
+		Where(m["n"].Const && m["n"].Text.Matches(`(\-[0-9]+|0)`)).
+		Suggest(`strings.ReplaceAll($s, $d, $w)`)
+	m.Match(`bytes.Replace($s, $d, $w, $n)`).
+		Where(m["n"].Const && m["n"].Text.Matches(`(\-[0-9]+|0)`)).
+		Suggest(`bytes.ReplaceAll($s, $d, $w)`)
+	m.Match(`strings.SplitN($s, $p, $n)`).
+		Where(m["n"].Const && m["n"].Text.Matches(`(\-[0-9]+|0)`)).
+		Suggest(`strings.Split($s, $p)`)
+	m.Match(`bytes.SplitN($s, $p, $n)`).
+		Where(m["n"].Const && m["n"].Text.Matches(`(\-[0-9]+|0)`)).
+		Suggest(`bytes.Split($s, $p)`)
+	m.Match(`strings.SplitAfterN($s, $p, $n)`).
+		Where(m["n"].Const && m["n"].Text.Matches(`(\-[0-9]+|0)`)).
+		Suggest(`strings.SplitAfter($s, $p)`)
+	m.Match(`bytes.SplitAfterN($s, $p, $n)`).
+		Where(m["n"].Const && m["n"].Text.Matches(`(\-[0-9]+|0)`)).
+		Suggest(`bytes.SplitAfter($s, $p)`)
+
+	// The last argument of `SplitN` indicates parts count, not splits count
+	m.Match(`strings.SplitN($s, $p, 1)`).Suggest(`strings.SplitN($s, $p, 2)`)
+	m.Match(`bytes.SplitN($s, $p, 1)`).Suggest(`bytes.SplitN($s, $p, 2)`)
+	m.Match(`strings.SplitAfterN($s, $p, 1)`).Suggest(`strings.SplitAfterN($s, $p, 2)`)
+	m.Match(`bytes.SplitAfterN($s, $p, 1)`).Suggest(`bytes.SplitAfterN($s, $p, 2)`)
+}
+
 func gocriticBadCall(m fluent.Matcher) {
-	m.Match(`strings.Replace($_, $_, $_, 0)`,
```

Comment:
same here - why removed?

Reply:
The rule is covered by the newly introduced more general rule. See the tests.